Sources/NetworkProtection/NetworkProtectionDeviceManager.swift

```diff
@@ -319,4 +317,22 @@ public actor NetworkProtectionDeviceManager: NetworkProtectionDeviceManagement {
 #endif
         errorEvents?.fire(clientError.networkProtectionError)
     }
+
+    private func getAccessToken() throws -> String {
+        do {
+            let token = try tokenStore.fetchToken()
+
+            if let token {
+                return token
+            } else {
+                throw NetworkProtectionError.noAuthTokenFound
+            }
+        } catch {
+            if let error = error as? AccountKeychainAccessError, error.isKeychainLockedError {
+                throw NetworkProtectionError.keychainLocked
```

Comment:
I believe it's ok as it is, but if there's ANY chance that this may ever be used for more than one underlying error, it's worth adding an associated error value for pixels to do their thing.